pkg/pulumiyaml/ast/template_test.go

```diff
@@ -118,3 +118,103 @@ func TestComponentParsing(t *testing.T) {
 	require.Len(t, template.Components.Entries[1].Value.Outputs.Entries, 1)
 	require.Equal(t, "bucketEndpoint", template.Components.Entries[1].Value.Outputs.Entries[0].Key.Value)
 }
+
+const componentSchemaExample = `
+name: yaml-plugin
+description: A YAML plugin
+runtime: yaml
+components:
+  aComponent:
+    description: "A component"
+    config:
+      someStringArray:
+        type: array
+        items:
+          type: string
+      aString:
+        type: string
+      aNumber:
+        type: integer
+      aBoolean:
+        type: boolean
+      aStringWithDefault:
+        type: string
+        default: "default"
+      aNumberWithDefault:
+        type: integer
+        default: 42
+      aBooleanWithDefault:
+        type: boolean
+        default: true
+    resources:
+      myBucket:
+        type: aws:s3/bucket:Bucket
+    outputs:
+      someOutput: "abcd"
+`
+
+func TestComponentSchemaGeneration(t *testing.T) {
+	t.Parallel()
+
+	syntax, diags := encoding.DecodeYAML("<stdin>", yaml.NewDecoder(strings.NewReader(componentSchemaExample)), nil)
+	require.Len(t, diags, 0)
+
+	template, diags := ParseTemplate([]byte(componentSchemaExample), syntax)
+	require.Len(t, diags, 0)
+
+	spec, err := template.GenerateSchema()
+	require.NoError(t, err)
+
+	require.Equal(t, "yaml-plugin", spec.Name)
```

Comment:
Probably a matter of taste, but I'd serialize to JSON and compare JSONs for easier readability